# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 46fd81530a8b
# Full Hash: 46fd81530a8b6d81b5520aff9451d49649f1420a
# Author: Ziran Sun <zsun@igalia.com>
# Date: 2023-05-23 09:47:12
# Regressor Bug: 1832837
# File Overlap Count: 1
# Description:
#   Bug 1832837 - Remove the fullscreen element ready check on fullscreen element descendant. r=edgar
#   
#   https://fullscreen.spec.whatwg.org/#fullscreen-element-ready-check
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D177913
# ==============================================================================

diff -r 069f4fb8fd87 -r 46fd81530a8b dom/base/Document.cpp
--- a/dom/base/Document.cpp	Mon May 22 11:53:20 2023 +0000
+++ b/dom/base/Document.cpp	Mon May 22 12:20:47 2023 +0000
@@ -15286,15 +15286,6 @@
     aRequest.Reject("FullscreenDeniedHTMLDialog");
     return false;
   }
-  // XXXsmaug Note, we don't follow the latest fullscreen spec here.
-  //         This whole check could be probably removed.
-  if (fullscreenElement && !nsContentUtils::ContentIsHostIncludingDescendantOf(
-                               elem, fullscreenElement)) {
-    // If this document is fullscreen, only grant fullscreen requests from
-    // a descendant of the current fullscreen element.
-    aRequest.Reject("FullscreenDeniedNotDescendant");
-    return false;
-  }
   if (!nsContentUtils::IsChromeDoc(this) && !IsInFocusedTab(this)) {
     aRequest.Reject("FullscreenDeniedNotFocusedTab");
     return false;